contracts/Distributor.sol

```diff
@@ -0,0 +1,250 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/utils/Context.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+
+contract Distributor is Context, Ownable {
+
+    uint256 private _numRegisteredForETH; // Counter to track no. of registered payees for ETH
+    uint256 private _totalETHShares;
+    uint256 private _totalETHDistributed;
+
+
+     /*///////////////////////////////////////////////////////////////
+                            Mappings
+    //////////////////////////////////////////////////////////////*/
+   
+
+    mapping(uint256 => address) private _payeesForETH;
+    // Specification of total share for ETHER per wallet
+    mapping(address => uint256) private _sharesForETH;
+    // Amount of ETH paid out to the account
+    mapping(address => uint256) private _distributedETH;
+
+    mapping(IERC20 => uint256) private _numRegisteredForToken; // Counter for each ERC20 Token to track no. of registered payees
+    mapping(IERC20 => mapping(uint256 => address)) private _payeesForToken; // IERC20 token -> index -> registered payee
+
+    mapping(IERC20 => uint256) private _totalTokenShares;
+    mapping(IERC20 => uint256) private _totalTokenDistributed;
+    // Specification of total shares tagged to the ERC20 Token Contract Address per wallet
+    mapping(IERC20 => mapping(address => uint256)) private _sharesForTokens; // IERC20 token -> account -> share
+    // Amount of Token specified paid out to the account
+    mapping(IERC20 => mapping(address => uint256)) private _distributedTokens; // IERC20 token -> account -> distributed Tokens
+
+        /*///////////////////////////////////////////////////////////////
+                            Events
+    //////////////////////////////////////////////////////////////*/
+    event RegisterPayeeForETH(address account, uint256 shares);
+    event RegisterPayeeForToken(address account, IERC20 token, uint256 shares );
+ 
+    event ETHFundClaimed(address account,uint256 shares);
+    event TokenFundClaimed(address account, IERC20 token, uint256 shares);
+
+    event ETHReceived(address from, uint256 amount);
+    event TokenReceived(address from, IERC20 token, uint256 amount);
+
+
+           /*///////////////////////////////////////////////////////////////
+                            Constructor
+    //////////////////////////////////////////////////////////////*/
+    // @Note: SHare 
+    constructor(address[] memory accounts_, uint256[] memory shares_) payable {
+        require(accounts_.length == shares_.length, "Distributor: Mismatch of addresses and shares.");
+        for(uint256 i = 0; i < accounts_.length; i++ ){
+            _registerPayeesForETH(accounts_[i], shares_[i]);
+        }
+    }
```

Comment:
Why you only distribute shares for eth in the constructor =)))
You can do the same thing with erc20, assign in the constructor, or just do both outside the constructor.
this is weird to me since it is not consistent

Reply:
Haha yeah it may be abit weird, but my primary consideration was to only allow the owner to register payees with their respective shares once for each ERC20 token, starting with the native token ETH as the constructor so yeah. I could add in features to allow amendments to shares and addition of payees too